umake/frameworks/ide.py

```diff
@@ -473,3 +474,83 @@ def post_install(self):
                                                      categories=categories))
         if not self.was_in_arduino_group:
             UI.delayed_display(DisplayMessage(_("You need to logout and login again for your installation to work")))
+            
+class BaseNetBeans(umake.frameworks.baseinstaller.BaseInstaller):
+    """The base for all Netbeans installers."""
+    
+    BASE_URL = "http://download.netbeans.org/netbeans/8.0.2/final/"
+
+    def __init__(self, category, version="8.0.2", flavour=""):
+        #add a separator to the string, like -cpp
+        if flavour != "":
+            flavour = '-' + flavour
+        super().__init__(name="Netbeans",
+                         description=_("Netbeans IDE " + version),
+                         category=category, 
+                         only_on_archs=['i386', 'amd64', 'arm'],
+                         download_page=None,
+                         dir_to_decompress_in_tarball="netbeans" + flavour,
+                         desktop_filename="netbeans" + flavour + ".desktop",
```

Comment:
I would prefer that you follow how we mostly do string manipulation in umake, that would be in those cases:
dir_to_decompress_in_tarball="netbeans{}".format(flavour),
desktop_filename="netbeans{}.desktop".format(flavour),

Please follow the same pattern in other string manipulation below, just for the sake of consistency :)

I'm not an english native, but I think it's more "flavor" than "flavour", right?
Also, we support wildcard for dir decompression, so, if there is only one directory in the tarball, you can use something like:
dir_to_decompress_in_tarball="netbeans*"